### ISSUE TITLE
SA1013 misfires for indexer expression

### ISSUE THREAD

**Issue by madelson** [DotNetAnalyzers/StyleCopAnalyzers#3037]
I was style-copping some code which selected a random value using the following syntax:

```
var choice = new[] { option1, option2, option3 }[random.Next(3)];
```

This triggers SA1013, which wants me to add whitespace after `}`. Of course, adding that whitespace triggers SA1010 (`[` must not be preceded by space). 

In this case, I believe that it is SA1013 that is in error; `}` should be allowed to be followed by `[`.

Thanks so much for maintaining this great library!

**Comment by vweijsters** [comment]
I'm not sure if we want to fix this. I'm somewhat temped to propose a new rule that would prevent object initializers from being directly followed by an indexer expression, as the readability of such a construct is questionable.
